e2e/deploy-vault.go

```diff
@@ -91,3 +97,79 @@ func createORDeleteVault(action string) {
 		e2elog.Failf("failed to %s vault psp %v", action, err)
 	}
 }
+
+// waitForJobCompletion polls the status of the given job and waits until the
+// jobs has succeeded or until the timeout is hit.
+func waitForJobCompletion(c kubernetes.Interface, ns, job string, timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+
+	e2elog.Logf("waiting for Job %s/%s to be in state %q", ns, job, batch.JobComplete)
+
+	return wait.PollImmediate(poll, t, func() (bool, error) {
+		j, err := c.BatchV1().Jobs(ns).Get(context.TODO(), job, metav1.GetOptions{})
+		if err != nil {
+			if isRetryableAPIError(err) {
+				return false, nil
+			}
+			return false, fmt.Errorf("failed to get Job: %w", err)
+		}
+
+		if j.Status.CompletionTime != nil {
+			// Job has successfully completed
+			return true, nil
+		}
+
+		e2elog.Logf(
+			"Job %s/%s has not completed yet (%d seconds elapsed)",
+			ns, job, int(time.Since(start).Seconds()))
+		return false, nil
+	})
+}
+
+// createTenantServiceAccount uses the tenant-sa.yaml example file to create
+// the ServiceAccount for the tenant and configured Hashicorp Vault with a
+// kv-store that the ServiceAccount has access to.
+func createTenantServiceAccount(c kubernetes.Interface, ns string) {
+	createORDeleteTenantServiceAccount("create", ns)
+
+	// wait for the Job to finish
+	const jobName = "vault-tenant-sa"
+	err := waitForJobCompletion(c, cephCSINamespace, jobName, deployTimeout)
+	if err != nil {
+		e2elog.Failf("Job %s/%s did not succeed: %v", cephCSINamespace, jobName, err)
```

Comment:
return an error from here and let the caller handle it?

Reply:
done!